jakarta-eclipse/src/org/jakarta/jdt/ServletDiagnosticsCollector.java

```diff
@@ -46,11 +53,36 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
 					}
 
 					if (isWebServletAnnotated && !isHttpServletExtended) {
-						ISourceRange nameRange = JDTUtils.getNameRange(type);
-						Range range = JDTUtils.toRange(unit, nameRange.getOffset(), nameRange.getLength());
+
 						diagnostics.add(new Diagnostic(range, "Classes annotated with @WebServlet must extend the HttpServlet class."));
 					}
 					
+					/* URL pattern diagnostic check */
+					if(WebServletAnnotation != null) {
+						IMemberValuePair[] memberValues = WebServletAnnotation.getMemberValuePairs();
+						
+						boolean isUrlpatternSpecified = false;
+						boolean isValueSpecified = false;
+						for (IMemberValuePair mv : memberValues) {
+							if(mv.getMemberName().equals(ServletConstants.URL_PATTERNS)) {
+								isUrlpatternSpecified = true;
+								continue;
+							}
+							if(mv.getMemberName().equals(ServletConstants.VALUE)) {
+								isValueSpecified = true;
+							}
+							
+						}
+						
+						if (!isUrlpatternSpecified && !isValueSpecified) {
+							diagnostics.add(new Diagnostic(range, "The annotated servlet MUST specify at least one url pattern to be deployed."));
```

Comment:
Same comments as above, diagnostic should appear on the class name and message should include the name of the annotation you are referring to